pkg/vsphere/compute/rp.go

```diff
@@ -88,3 +88,51 @@ func (rp *ResourcePool) GetCluster(ctx context.Context) (*object.ComputeResource
 
 	return object.NewComputeResource(rp.Client.Client, mrp.Owner), nil
 }
+
+func (rp *ResourcePool) GetDatacenter(ctx context.Context) (*object.Datacenter, error) {
+	dcRef, err := rp.getLowestAncestor(ctx, "Datacenter")
+
+	if err != nil || dcRef == nil {
+		log.Errorf("Unable to get datacenter ancestor of rp %s: %s", rp.Name(), err)
+		return nil, err
+	}
+
+	return object.NewDatacenter(rp.Client.Client, *dcRef), nil
+}
+
+func (rp *ResourcePool) getAncestors(ctx context.Context, inType string) ([]types.ManagedObjectReference, error) {
+	client := rp.Session.Vim25()
+
+	ancestors, err := mo.Ancestors(ctx, client, client.ServiceContent.PropertyCollector, rp.Reference())
+
+	if err != nil {
+		log.Errorf("Unable to get ancestors of rp %s: %s", rp.Name(), err)
+		return nil, err
+	}
+
+	outAncestors := make([]types.ManagedObjectReference, 0)
```

Comment:
you may as well make the last parameter to make be the maximum length - just polite as Go has (had?) issues with memory fragmentation.
```
make([]types.ManagedObjectReference, 0, len(ancestors)) 
```

Reply:
Done